pkg/sync/grpc/grpc_sync.go

```diff
@@ -0,0 +1,186 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/credentials/insecure"
+
+	"buf.build/gen/go/open-feature/flagd/grpc/go/sync/v1/syncv1grpc"
+	v1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/sync/v1"
+
+	"github.com/open-feature/flagd/pkg/logger"
+	"github.com/open-feature/flagd/pkg/sync"
+	"google.golang.org/grpc"
+)
+
+const (
+	// Prefix for GRPC URL inputs. GRPC does not define a prefix through standard. This prefix helps to differentiate
+	// remote URLs for REST APIs (i.e - HTTP) from GRPC endpoints.
+	Prefix = "grpc://"
+
+	// Connection retry constants
+	// Back off period is calculated with backOffBase ^ #retry-iteration. However, when #retry-iteration count reach
+	// backOffLimit, retry delay fallback to constantBackOffDelay
+	backOffLimit         = 3
+	backOffBase          = 4
+	constantBackOffDelay = 60
+)
+
+type Sync struct {
+	Target     string
+	ProviderID string
+	Logger     *logger.Logger
+}
+
+func (g *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
```

Comment:
do you think, maybe in another PR, we could allow users to pass their own `DialOption` and fallback on insecure transport if none provided ?

cause since flagd may be used in a distributed environment (k8s pods sidecars), I would for example need to add [OTEL tracing instrumentation](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/google.golang.org/grpc/otelgrpc/example/client/main.go#L46-L49) and users could tweak depending on their needs

Reply:
I am actively working on extending grpc connectivity to support TLS connections [1]. This should allow altering connectivity between TLS and Insecure based on schema (`grpc://` vs `grpcs://`). Further, we are having high-level discussions/investigations on improving configuration options of flagd [2].

However, at least for now, there is no plan to support OTEL tracing and related configurations. If you see this as a valid use case, if you have some spare time, you can always contribute with a PR. Our community is happy to help you here. 

[1] - https://github.com/open-feature/flagd/pull/398
[2] - https://github.com/open-feature/flagd/issues/405